openupgradelib/openupgrade_merge_records.py

```diff
@@ -421,6 +425,23 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
             if op == 'merge':
                 if not getattr(target_record, field.name) and _list:
                     vals[field.name] = _list[0]
+        elif field.type == 'many2one_reference':
+            if field.model_field not in model._fields:
+                continue
+            model_field = [x for x in fields if x.name == field.model_field][0]
```

Comment:
No need to twist the thing:
```suggestion
            model_field = model._fields[field.model_field]
```